sksurv/tree/_criterion.pyx

```diff
@@ -315,13 +315,14 @@ cdef class LogrankCriterion(Criterion):
         dest[0] = ratio  # Nelson-Aalen estimator
         dest[1] = 1.0 - ratio  # Kaplan-Meier estimator
 
-        j = 2
+        j_delta = 3
+        j = j_delta
         for k in range(1, self.n_event_times):
             rs = &self.riskset_total[k]
-            dest[j] = dest[j - 2]
-            dest[j + 1] = dest[j - 1]
+            dest[j] = dest[j - j_delta]
```

Comment:
`dest` is a pointer to the array that hold the predictions. It's a 2d array of size `n_unique_times × 2`. The first column is the survival function, the second column the cumulative hazard function. This is also why incrementing by 2 is required.

In low-memory mode, `dest` should have `shape = (2,)` by changing `SurvivalTree.n_classes_` and `SurvivalTree.n_outputs_` accordingly. Then, `dest[0]` should hold the sum over the CHF, and `dest[1]` the sum over the survival function.

Note that the code is going to change with #371. Once it is merged, CHF and survival function will be over all unique time points (not just event times). Thus, the summing operation also needs to consider at which times an event was observed.

Reply:
Thanks @sebp for the pointers here!

I've made changes and it almost works now, only the _"the summing operation also needs to consider at which times an event was observed"_ bit remains to be figured out.